lib/plugins/bed.js

```diff
@@ -1,8 +1,38 @@
+const Vec3 = require('vec3').Vec3
+
 module.exports = inject
 
 function inject (bot) {
   bot.isSleeping = false
 
+  function parseBedMetadata (bedBlock) {
+    const metadata = {}
+    const bMetadata = bedBlock.metadata.toString(2).padStart(4, '0')
+
+    metadata.part = Number(bMetadata[0])
+    metadata.occupied = Number(bMetadata[1])
```

Comment:
This will be a bit faster in V8:
```js
const metadata = {
  part: // ...
  occupied: // ...
}
```

Also add the default outcome (judging by L10 it's `00`) to the object at creation time and then simply modify the string and update the vector without creating a new instance

And my personal preference is to use `+"8"` rather than `Number` or `parseInt`. Less writing and kind of understandable if you know how javascript works

Reply:
I don't see what the `+"8"` would do
Thanks for the feedback btw!

Reply:
Oh, it's just a quick way to convert to a number